packages/fast-usdc/src/utils/fees.js

```diff
@@ -0,0 +1,61 @@
+import { AmountMath } from '@agoric/ertp';
+import { multiplyBy } from '@agoric/zoe/src/contractSupport/ratio.js';
+import { Fail } from '@endo/errors';
+import { mustMatch } from '@endo/patterns';
+import { FeeConfigShape } from '../typeGuards.js';
+
+const { add, subtract, isEqual, isGTE } = AmountMath;
+
+/**
+ * @import {Amount, Brand} from '@agoric/ertp';
+ * @import {FeeConfig} from '../types.js';
+ */
+
+/**
+ * @param {Brand} brand
+ * @param {FeeConfig} feeConfig
+ */
+export const makeFeeTools = (brand, feeConfig) => {
+  mustMatch(feeConfig, FeeConfigShape, 'Must provide feeConfig');
+  const { flat, variableRate, maxVariable } = feeConfig;
+  const emptyAmount = AmountMath.makeEmpty(brand);
+  return harden({
+    /**
+     * Calculate the net amount to advance after withholding fees.
+     *
+     * @param {Amount<'nat'>} requested
+     */
+    calculateAdvance(requested) {
+      const fee = this.calculateAdvanceFee(requested);
+      return subtract(requested, fee);
+    },
+    /**
+     * Calculate the total fee to charge for the advance.
+     *
+     * @param {Amount<'nat'>} requested must be greater than 0
+     */
+    calculateAdvanceFee(requested) {
+      !isEqual(requested, emptyAmount) || Fail`Fee exceeds requested.`;
+      const potentialVariable = multiplyBy(requested, variableRate);
+      const variable = isGTE(potentialVariable, maxVariable)
+        ? maxVariable
+        : potentialVariable;
```

Comment:
`AmountMath` has `min` and `max`, so:

```suggestion
      const variable = min(multiplyBy(requested, variableRate), maxVariable);
```

Reply:
Awesome, thanks